libs/barista-components/expandable-text/src/expandable-text.ts

```diff
@@ -45,6 +51,9 @@ import { Observable } from 'rxjs';
   encapsulation: ViewEncapsulation.Emulated,
 })
 export class DtExpandableText {
+  /** @internal Unique identifier on the page for the expandable text. */
```

Comment:
Same here i think for the id - maybe a mixin makes sense that takes a string for the uniqueid generation?

Reply:
I'm not sure how I would be able to do this in a mixin, as the element should get an ID that is referring somewhat to the component name or the selector.

But I can give it a try. One thing I can already see is, that it will be far less readable.